### DISCUSSION TITLE
Homework 2 Part 1.1

### DISCUSSION THREAD

**Discussion by yams11** [cooperrc/computational-mechanics#37] [Q&A]
Hey! Do any of y'all know how to get the "!head" function to read the file? I opened the URL and wrote this, but I am not sure what else to do so python can read the data.
<img width="806" alt="Screen Shot 2022-06-07 at 1 41 49 PM" src="https://user-images.githubusercontent.com/106359653/172447869-d24878c7-0cf3-4d71-a45d-b1c2c4980d1a.png">
)

**Comment by Aarons-Joshua** [comment]
Have you tried running the code below?

from urllib.request import urlretrieve
URL = 'http://go.gwu.edu/engcomp2data1?accessType=DOWNLOAD'
urlretrieve(URL, 'beers.csv')

**Reply by Aarons-Joshua** [reply]
![image](https://user-images.githubusercontent.com/98834673/172529024-c28b5761-5e92-4a65-b207-0f971f90372c.png)

**Comment by cooperrc** [comment]
If you're getting the `No such file or directory` the file might be in a different location (for example if you are working on a notebook in the `~/work` folder, but the file is in the `~/computational-mechanics/data` folder)

@Aarons-Joshua workaround is good way to find a persistent link location if you don't want to search the home folder+work folder.

**Reply by yams11** [reply]
@cooperrc,
How do I access the file? I do not know where it is.

**Reply by cooperrc** [reply]
Depends on the way you're working on your HW:
### 1. You clicked the rocket icon to "Launch in JupyterHub"
- Your HW file is located in `~/computational-mechanics/module_02` and the data is in `~/computational-mechanics/data`
- access the file with `!head ../data/transistor_data.csv`
### 2. You clicked the rocket icon to "Launch in JupyterHub" then clicked "Save as" -> `/work/HW_02.md`
- Your HW file is located in `~/work` and the data is in `~/computational-mechanics/data` 
- access the file with `!head ~/computational-mechanics/data/transistor_data.csv`
### 3. You used `git clone https:/github.com/username/computational-mechanics`
- Your HW file is located in `~/work/computational-mechanics/module_02` and the data is in `~/work/computational-mechanics/data`
- access the file with `!head ../data/transistor_data.csv`
